diff --git a/dist/cjs/util.js b/dist/cjs/util.js
index 77e64b56872bc65bcb0f60ecfbec84eb3da02c6f..f79a25512dd1d44e6ff08793b9f70f8ad51f1e28 100644
--- a/dist/cjs/util.js
+++ b/dist/cjs/util.js
@@ -151,7 +151,7 @@ async function walkAsync(dirPath) {
     (0, exports.debugLog)('Walking... ' + dirPath);
     async function _walkAsync(dirPath) {
         const children = await fs.readdir(dirPath);
-        return await Promise.all(children.map(async (child) => {
+        const walkChild = async (child) => {
             const filePath = path.resolve(dirPath, child);
             const stat = await fs.stat(filePath);
             if (stat.isFile()) {
@@ -174,7 +174,12 @@ async function walkAsync(dirPath) {
                 return walkResult;
             }
             return null;
-        }));
+        };
+        const results = [];
+        for (const child of children) {
+            results.push(await walkChild(child));
+        }
+        return results;
     }
     const allPaths = await _walkAsync(dirPath);
     return compactFlattenedList(allPaths);
diff --git a/dist/esm/util.js b/dist/esm/util.js
index d3f66d54b1a88e7410d78affe17de87302db02a3..8057cf773e0350951b6faefed7a21d80e47cedc2 100644
--- a/dist/esm/util.js
+++ b/dist/esm/util.js
@@ -115,7 +115,7 @@ export async function walkAsync(dirPath) {
     debugLog('Walking... ' + dirPath);
     async function _walkAsync(dirPath) {
         const children = await fs.readdir(dirPath);
-        return await Promise.all(children.map(async (child) => {
+        const walkChild = async (child) => {
             const filePath = path.resolve(dirPath, child);
             const stat = await fs.stat(filePath);
             if (stat.isFile()) {
@@ -138,7 +138,12 @@ export async function walkAsync(dirPath) {
                 return walkResult;
             }
             return null;
-        }));
+        };
+        const results = [];
+        for (const child of children) {
+            results.push(await walkChild(child));
+        }
+        return results;
     }
     const allPaths = await _walkAsync(dirPath);
     return compactFlattenedList(allPaths);
